routers/web/repo/repo.go

```diff
@@ -443,6 +444,7 @@ func download(ctx *context.Context, archiveName string, archiver *repo_model.Rep
 		//If we have a signed url (S3, object storage), redirect to this directly.
 		u, err := storage.RepoArchives.URL(rPath, downloadName)
 		if u != nil && err == nil {
+			addCommitObjectResponseHeader(ctx, archiver)
```

Comment:
In my mind, we do not need the response header here, the commit-related header should be set in the redirected response, HTTP client follows that location.

Reply:
Then how HTTP client works? Users should parse the first 30x response to get a commit ID, and then read the redirected response to get content? It seems very inconvenient and strange.